### ISSUE TITLE
Autocomplete for CommonJS (const, require) does not exist

### ISSUE THREAD

**Issue by ChristianElijah** [fastify/fastify#3056]
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure it has not already been reported

### Fastify version

3.15.1

### Node.js version

14.16.0

### Operating system

macOS

### Operating system version (i.e. 20.04, 11.3, 10)

11.2.3

### Description

Hello everyone, my name is Christian

I have been having this common problem where intellisense/autocomplete doesn't work for common JS, yet it works for ESM. 

// Common JS
![Screen Shot 2021-05-05 at 9 45 38 AM](https://user-images.githubusercontent.com/83177759/117140005-6f094200-ad87-11eb-9f60-628e099a5c7d.png)

// ESM
![Screen Shot 2021-05-05 at 9 46 54 AM](https://user-images.githubusercontent.com/83177759/117140020-76305000-ad87-11eb-95e4-7349bd9d3cf8.png)

Is there some config i need to setup before hand?

### Steps to Reproduce

`const person = async (fastify, opts) => {
  fastify.get('/', async function(request, reply) {
      reply.code(200).send({
        success: true,
        data: 'All People Retrieved'
      });
  });

  fastify.post('/', async function(request, reply) {
      reply.code(200).send({
        success: true,
        data: 'Added a new person'
      });
  });
  
}

module.exports = person;`

**Comment by RafaelGSS** [comment]
If you import using: `require('fastify').default` works, right?

**Comment by ChristianElijah** [comment]
> If you import using: `require('fastify').default` works, right?

Yes it does but is there a way for app wide?

**Comment by RafaelGSS** [comment]
Not sure what you mean with _app-wide_. The example that _"not work"_ using commonjs is not requiring `fastify` in any place, so the IntelliSense doesn't know what is the type of this parameter.

To fix it using javascript, you can use js notation
```
/**
 * @param {import("fastify").FastifyInstance} fastify 
 * @param {*} opts 
 */
module.exports = async function (fastify, opts) {
  fastify.get('/look', () => 'at me');
}
```

Further details: https://github.com/fastify/fastify/blob/main/docs/TypeScript.md#code-completion-in-vanilla-javascript

**Comment by RafaelGSS** [comment]
We had several discussions with that in some issues. @fox1t Do you know if we have some document or issue to explain this behavior?

**Comment by climba03003** [comment]
> 
> 
> Couldnt this approach be used to satisfy both those who use commonjs and those who use esmodules?
> (https://remarkablemark.org/blog/2020/05/05/typescript-export-commonjs-es6-modules/)
> TLDR
> using this on the d.ts
> 
> ```
> myModule.default = myModule;
> export = myModule;
> ```
> 
> instead of
> 
> ```
> export default myModule;
> ```
> 
> I'm not familiar with the codebase to make a pull request, but im having the same issue trying to use commonjs with fastify, the .default works but it looks like it breaks the pattern

We need to support typing for.
```js
import { fastify } from 'fastify' // it will break
import fastify from 'fastify'
const fastify = require('fastify')
const { fastify } = require('fastify')
```
Your example will be works for some of them, but breaking most of them.

**Comment by JoCat** [comment]
> > If you import using: `require('fastify').default` works, right?
> 
> Yes it does but is there a way for app wide?

`const { fastify } = require("fastify");` works for me

**Comment by climba03003** [comment]
As the current typing system and IntelliSense rely on TypeScript typing.
It is impossible to support both `CJS` and `ESM`.
